api/envoy/config/core/v3/address.proto

```diff
@@ -119,6 +119,19 @@ message TcpKeepalive {
 message ExtraSourceAddress {
   // The additional address to bind.
   SocketAddress address = 1 [(validate.rules).message = {required: true}];
+
+  // [#not-implemented-hide:]
+  // Additional socket options that may not be present in Envoy source code or
+  // precompiled binaries. If no socket options specified, the BindConfig's
+  // :ref:`socket_options <envoy_v3_api_field_config.core.v3.BindConfig.socket_options>`
+  // will apply to this address.
+  repeated SocketOption socket_options = 2;
```

Comment:
Would prefer extract this to a message as `socket_options_override`, then you don't need `no_socket_options`. If the field is `null` means no override, non-null with empty means no socket options will be applied.

Reply:
got it, thanks! that is good idea